jaeger_client/codecs.py

```diff
@@ -82,7 +82,7 @@ def inject(self, span_context, carrier):
                 carrier[header_key] = encoded_value
 
     def extract(self, carrier):
-        if not hasattr(carrier, 'iteritems'):
+        if not hasattr(carrier, 'items'):
```

Comment:
I am not able to find the reason why this code was checking for `iteritems` instead of `isinstance(carrier, dict)` as done in the inject method, but I seem to recall there was a reason for it, i.e. a type was passed in some cases that wasn't a true dict but allowed iteritems (with there was a unit test).

I would feel safer if this check was backwards compatible, i.e. check for `items` OR `iteritems`.